test/lib/client.test.js

```diff
@@ -196,6 +200,62 @@ describe('Node AMQP Bus Client', function testBus() {
         ackStub.calledOnce.should.be.true();
         fakeHandler.called.should.be.false();
       });
+
+      describe('graceful shutdown', () => {
+        beforeEach(() => {
+          // without this, mocha will stop at the end of the test
+          process.removeAllListeners('SIGTERM');
+          process.removeAllListeners('SIGINT');
+        });
+
+        it('should stop consuming messages when receiving SIGINT', function* it() {
+          yield busClient.setupQueue(exchange, queue, rootingKey);
+          const cancelSpy = sandbox.spy(busClient.channel, 'cancel');
+
+          function* handler(content, fields, callback) {
+            fakeHandler(content);
+            callback(new Error());
+          }
+
+          yield busClient.consume(queue, handler);
+
+          process.kill(process.pid, 'SIGINT');
+
+          yield cb => setTimeout(cb, 50);
+
+          cancelSpy.calledOnce.should.be.true();
```

Comment:
should 😱

Reply:
normally we have expect at the front (instead of cancelSpy directly) is there a reason why it's different in this project?